diagnosticsModule/Private/AdfsHealthChecks.ps1

```diff
@@ -361,25 +361,39 @@ Function TestADFSDuplicateSPN
         }
 
         # Verify User is either in domain\user format or UPN format, get the domain and username values and make Service account is in the domain\user format.
-        $serviceAccountParts = $adfsServiceAccount.Split('\\')
-        if ($serviceAccountParts.Length -ne 2)
+        if (IsUserPrincipalNameFormat($adfsServiceAccount))
         {
             $serviceAccountPartsUpn = $adfsServiceAccount.Split('@')
             if ($serviceAccountPartsUpn.Length -ne 2)
             {
                 throw "Unexpected value of the service account $adfsServiceAccount. Expected in DOMAIN\\User format or UPN:User@Domain"
```

Comment:
The error message is a little misleading. The format you expect is here the UPN one. IsUserPrincipalNameFormat() won't allow domain\user, right? . As mentioned in comment below, you could chose to split and pass or have the callee split. I prefer latter because it is useful elsewhere.